cnf-certification-test/accesscontrol/suite.go

```diff
@@ -410,44 +410,58 @@ func testPodRoleBindings(env *provider.TestEnvironment) {
+				// Determine if that service account is not in the same namespace as the pod.
+				if subject.Kind != "ServiceAccount" || subject.Name != put.Spec.ServiceAccountName {
+					continue
+				}
 
-		if len(roleBindings) > 0 {
-			logrus.Warnf("Pod: %s/%s has the following role bindings: %s", put.Namespace, put.Name, roleBindings)
-			tnf.ClaimFilePrintf("Pod: %s/%s has the following role bindings: %s", put.Namespace, put.Name, roleBindings)
-			failedPods = append(failedPods, put.Name)
+				if rbac.IsRoleBindingOutOfNamespace(put.Namespace, put.Spec.ServiceAccountName, env.RoleBindings[rbIndex].Namespace, env.RoleBindings[rbIndex].Name) {
+					logrus.Warnf("Pod: %s/%s has the following role bindings that do not live in the same namespace: %s", put.Namespace, put.Name, env.RoleBindings[rbIndex].Name)
+					tnf.ClaimFilePrintf("Pod: %s/%s has the following role bindings that do not live in the same namespace: %s", put.Namespace, put.Name, env.RoleBindings)
+					failedPods = append(failedPods, put.Name)
+					found = true
+					break
+				}
+			}
+			// Break of out the loop if we found a role binding that is out of namespace
+			if found {
+				break
+			}
 		}
 	}
 	testhelper.AddTestResultLog("Non-compliant", failedPods, tnf.ClaimFilePrintf, ginkgo.Fail)
 }
 
-// testPodClusterRoleBindings verifies that the pod utilizes a valid cluster role binding that does not cross namespaces
+// testPodClusterRoleBindings verifies that the pod does not use a cluster role binding
 //
 //nolint:dupl
 func testPodClusterRoleBindings(env *provider.TestEnvironment) {
-	ginkgo.By("Should not have ClusterRoleBinding in other namespaces")
+	ginkgo.By("Pods should not have ClusterRoleBindings")
 	failedPods := []string{}
 
+	logrus.Infof("There were %d cluster role bindings found in the cluster.", len(env.ClusterRoleBindings))
+
 	for _, put := range env.Pods {
 		ginkgo.By(fmt.Sprintf("Testing cluster role binding for pod: %s namespace: %s", put.Name, put.Namespace))
-		if put.Spec.ServiceAccountName == "" {
-			logrus.Infof("%s has an empty serviceAccountName, skipping.", put.String())
-			continue
-		}
-
-		// Get any clusterrolebindings that do not belong to the pod namespace.
-		clusterRoleBindings, err := rbac.GetClusterRoleBindings(put.Namespace, put.Spec.ServiceAccountName)
+		result, err := put.IsUsingClusterRoleBinding(env.ClusterRoleBindings)
 		if err != nil {
+			logrus.Errorf("failed to determine if pod %s/%s is using a cluster role binding: %v", put.Namespace, put.Name, err)
 			failedPods = append(failedPods, put.Name)
 		}
 
-		if len(clusterRoleBindings) > 0 {
-			logrus.Warnf("Pod: %s/%s has the following cluster role bindings: %s", put.Namespace, put.Name, clusterRoleBindings)
-			tnf.ClaimFilePrintf("Pod: %s/%s has the following cluster role bindings: %s", put.Namespace, put.Name, clusterRoleBindings)
+		// Pod was found to be using a cluster role binding.  This is not allowed.
+		// Flagging this pod as a failed pod.
+		if !result {
```

Comment:
The condition here should be `if result`, and not `if !result`, shouldn't it?
Because, in the function called `IsUsingClusterRoleBinding`, it returns `true` when a pod is using a ClusterRoleBinding, `else` if not using it. And here, we are checking the opposite.
That's why the DCI job is failing: in fact, we've removed all the ClusterRoleBindings and ClusterRoles from the pods under test, and despite that, the job is failing in the access-control-cluster-role-bindings test. And it's because of this condition.

Reply:
Fixed!